sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/DiagnosticsProvider.java

```diff
@@ -667,15 +670,120 @@ void addEvent(String name, Map<String, Object> attributes, OffsetDateTime timest
     }
 
     private final static class OpenTelemetryCosmosTracer implements CosmosTracer {
+        private final Tracer tracer;
+
+        public OpenTelemetryCosmosTracer(Tracer tracer) {
+            checkNotNull(tracer, "Argument 'tracer' must not be null.");
+            this.tracer = tracer;
+        }
 
         @Override
         public Context startSpan(String spanName, CosmosDiagnosticsContext cosmosCtx, Context context) {
-            return null;
+
+            checkNotNull(spanName, "Argument 'spanName' must not be null.");
+            checkNotNull(cosmosCtx, "Argument 'cosmosCtx' must not be null.");
+            Context local = Objects
+                .requireNonNull(context, "'context' cannot be null.")
+                .addData(COSMOS_DIAGNOSTICS_CONTEXT_KEY, cosmosCtx);
+
+            StartSpanOptions spanOptions = this.startSpanOptions(
+                spanName,
+                cosmosCtx);
+
+            // start the span and return the started span
+            return tracer.start(spanName, spanOptions, local);
+        }
+
+        private StartSpanOptions startSpanOptions(String spanName, CosmosDiagnosticsContext cosmosCtx) {
+            StartSpanOptions spanOptions = new StartSpanOptions(SpanKind.CLIENT)
+                .setAttribute(AZ_TRACING_NAMESPACE_KEY, RESOURCE_PROVIDER_NAME)
+                .setAttribute("db.system","cosmosdb")
+                .setAttribute("db.operation",spanName)
+                .setAttribute("net.peer.name",cosmosCtx.getAccountName());
```

Comment:
Based on https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/database/
net.peer.name should be Name of the database host (For direct mode spans, it should be Service Address host name)
Also, net.peer.port could be used to capture Service Address port number

Reply:
This trace is for SDK operation - not physical network request - so physical address doesn't work. We had that discussion with OTel folks on the .Net side - whether to use a different attribute name or use this as account. Feedback was to use the account - because when looking at the big picture - app calling downstream service which might each use Cosmos or other database account level is the right abstraction. There will be option to also add request-level tracing - that will have physical address of replica for direct mode.

Reply:
net.peer.name here will have the full hostname of the ServiceEndpoint - identifying the account + cloud - so, the Azure resource